packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -110,37 +148,55 @@ export function cloudflare<
 								type: 'ESModule',
 								path: runnerPath,
 							},
+							{
+								// we declare the workerd-custom-import as a CommonJS module, thanks to this
+								// require is made available in the module and we are able to handle cjs imports
+								type: 'CommonJS',
+								path: workerdCustomImportPath,
+								contents: 'module.exports = path => import(path)',
+							},
 						],
 						serviceBindings: {
 							...workerOptions.serviceBindings,
 							__VITE_FETCH_MODULE__: async (request) => {
-								const args = (await request.json()) as [
-									string,
-									string,
-									FetchFunctionOptions,
-								];
+								const [moduleId, imported, options] =
+									(await request.json()) as [
+										string,
+										string,
+										FetchFunctionOptions,
+									];
 
 								const devEnvironment = viteDevServer.environments[
 									workerOptions.name
 								] as CloudflareDevEnvironment;
 
 								try {
-									const result = await devEnvironment.fetchModule(...args);
+									const result = await devEnvironment.fetchModule(
+										moduleId,
+										imported,
+										options,
+									);
 
 									return new MiniflareResponse(JSON.stringify(result));
 								} catch (error) {
-									// TODO: check error handling
-									const result = {
-										externalize: args[0],
-										type: 'builtin',
-									} satisfies vite.FetchResult;
-
-									return new MiniflareResponse(JSON.stringify(result));
+									if (moduleId.startsWith('cloudflare:')) {
+										const result = {
+											externalize: moduleId,
+											type: 'module',
+										} satisfies vite.FetchResult;
+
+										return new MiniflareResponse(JSON.stringify(result));
+									}
+									throw new Error(
+										`Unexpected Error, failed to get module: ${moduleId}`,
+									);
 								}
 							},
 						},
 					};
 				}),
+				unsafeUseModuleFallbackService: true,
```

Comment:
Why is this twice, here and on line 140 above?

Reply:
left over from refactoring, sorry 🙇